src/logger/Logger.cpp

```diff
@@ -0,0 +1,194 @@
+#include "kompute/logger/Logger.hpp"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <mutex>
+#include <spdlog/async.h>
+#include <spdlog/common.h>
+#include <spdlog/logger.h>
+#include <spdlog/sinks/rotating_file_sink.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
+
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#ifdef _WIN32
+#include <iso646.h>
+#include <direct.h>
+#endif // _WIN32
+
+namespace logger {
+constexpr int THREAD_QUEUE_LENGTH = 8192;
+constexpr int FILE_ROTATION_TIME = 1048576 * 5;
+
+/**
+ * Should be replaced with:
+ * std::filesystem::exists(path)
```

Comment:
Is this something that should be encompassed in a marco guard?

Reply:
Ok I do see what you mean on this, I remember coming across this issue at some point - for file check this would make sense.